
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>GSVA Python CLI &#8212; GSVA Python CLI 1.0.0 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="gsva-python-cli">
<h1>GSVA Python CLI<a class="headerlink" href="#gsva-python-cli" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<span class="target" id="module-GSVA"></span><p>Execute bioconductors GSVA transformation of gene expression into pathway enrichment</p>
<dl class="function">
<dt id="GSVA.gsva">
<code class="descclassname">GSVA.</code><code class="descname">gsva</code><span class="sig-paren">(</span><em>expression_df</em>, <em>geneset_df=None</em>, <em>gmt_file=None</em>, <em>method='gsva'</em>, <em>kcdf='Gaussian'</em>, <em>abs_ranking=False</em>, <em>min_sz=1</em>, <em>max_sz=None</em>, <em>parallel_sz=0</em>, <em>parallel_type='SOCK'</em>, <em>mx_diff=True</em>, <em>tau=None</em>, <em>ssgsea_norm=True</em>, <em>verbose=False</em>, <em>tempdir=None</em><span class="sig-paren">)</span><a class="headerlink" href="#GSVA.gsva" title="Permalink to this definition">¶</a></dt>
<dd><p>GSVA function for use with pandas DataFrame objects</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>expression_df</strong> (<em>pandas.DataFrame</em>) – REQUIRED: Expression data indexed on gene names column labels as sample ids</li>
<li><strong>geneset_df</strong> (<em>pandas.DataFrame</em>) – REQUIRED: Genesets and their members in a dataframe</li>
<li><strong>gmt_file</strong> (<em>string</em>) – Optionally load genesets from a gmt_file</li>
<li><strong>method</strong> (<em>string Default: 'gsva'</em>) – Method to employ in the estimation of gene-set enrichment scores per sample. By default this is set to gsva (Hänzelmann et al, 2013) and other options 6 gsva are ssgsea (Barbie et al, 2009), zscore (Lee et al, 2008) or plage (Tomfohr et al, 2005). The latter two standardize first expression profiles into z-scores over the samples and, in the case of zscore, it combines them together as their sum divided by the square-root of the size of the gene set, while in the case of plage they are used to calculate the singular value decomposition (SVD) over the genes in the gene set and use the coefficients of the first right-singular vector as pathway activity profile.</li>
<li><strong>kcdf</strong> (<em>string Default: 'Gaussian'</em>) – Character string denoting the kernel to use during the non-parametric estimation of the cumulative distribution function of expression levels across samples when method=”gsva”. By default, kcdf=”Gaussian” which is suitable when input expression values are continuous, such as microarray fluorescent units in logarithmic scale, RNA-seq log-CPMs, log-RPKMs or log-TPMs. When input expression values are integer counts, such as those derived from RNA-seq experiments, then this argument should be set to kcdf=”Poisson”. This argument supersedes arguments rnaseq and kernel, which are deprecated and will be removed in the next release.</li>
<li><strong>abs_ranking</strong> (<em>bool Default: False</em>) – Flag used only when mx_diff=TRUE. When abs_ranking=FALSE [default] a modified Kuiper statistic is used to calculate enrichment scores, taking the magnitude difference between the largest positive and negative random walk deviations. When abs.ranking=TRUE the original Kuiper statistic that sums the largest positive and negative random walk deviations, is used. In this latter case, gene sets with genes enriched on either extreme (high or low) will be regarded as ’highly’ activated.</li>
<li><strong>min_sz</strong> (<em>int Default: 1</em>) – Minimum size of the resulting gene sets.</li>
<li><strong>max_sz</strong> (<em>int Default: Inf</em>) – Maximum size of the resulting gene sets. Leave unset for no limit.</li>
<li><strong>parallel_sz</strong> (<em>int Default: 0</em>) – Number of processors to use when doing the calculations in parallel. This requires to previously load either the parallel or the snow library. If parallel is loaded and this argument is left with its default value (parallel_sz=0) then it will use all available core processors unless we set this argument with a smaller number. If snow is loaded then we must set this argument to a positive integer number that specifies the number of processors to employ in the parallel calculation.</li>
<li><strong>parallel_type</strong> (<em>string Default: &quot;SOCK&quot;</em>) – Type of cluster architecture when using snow.</li>
<li><strong>mx_diff</strong> (<em>bool Default: True</em>) – Offers two approaches to calculate the enrichment statistic (ES) from the KS random walk statistic. mx_diff=FALSE: ES is calculated as the maximum distance of the random walk from 0. mx_diff=TRUE (default): ES is calculated as the magnitude difference between the largest positive and negative random walk deviations.</li>
<li><strong>tau</strong> (<em>float</em>) – Exponent defining the weight of the tail in the random walk performed by both the gsva (Hänzelmann et al., 2013) and the ssgsea (Barbie et al., 2009) methods. By default, this tau=1 when method=”gsva” and tau=0.25 when method=”ssgsea” just as specified by Barbie et al. (2009) where this parameter is called alpha. Leave unset for defaults.</li>
<li><strong>ssgsea_norm</strong> (<em>bool Default: True</em>) – Logical, set to TRUE (default) with method=”ssgsea” runs the SSGSEA method from Barbie et al. (2009) normalizing the scores by the absolute difference between the minimum and the maximum, as described in their paper. When ssgsea_norm=FALSE this last normalization step is skipped.</li>
<li><strong>verbose</strong> (<em>bool Default: False</em>) – Gives information about each calculation step.</li>
<li><strong>tempdir</strong> (<em>string Default: System Default</em>) – Location to write temporary files</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">pandas.DataFrame</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper"><div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="#">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, Jason L Weirather.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.6.7</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>